src/lib.rs

```diff
@@ -285,32 +292,194 @@ mod tests {
         ];
 
         #[rustfmt::skip]
-        let velocities = vec![
+        let values = vec![
             -9., -8., -7., -6., -5., -4.,
             -3., -2., -1.,  0.,  1.,  2.,
              3.,  4.,  5.,  6.,  7.,  8.,
         ];
 
-        // output buffers
-        let mut mean_vec = vec![0.0; 3];
-        let mut weight_vec = vec![0.0; 3];
+        (positions, values)
+    }
+
+    // apply_accum Tests
+    // -----------------
+
+    // based on numpy! (maybe look into a better approach)
+    // https://numpy.org/doc/stable/reference/generated/numpy.isclose.html
+    //
+    // maybe look into the approx crate?
```

Comment:
I would advocated for fewer dependencies.  This implementation seems more than adequate, unless our test suite becomes very sophisticated.

Reply:
Neat, that is a reasonable point.